Source/DafnyCore/AST/Attributes.cs

```diff
@@ -1,21 +1,352 @@
+      }
+      case "AutoRevealDependencies": {
+        return A1("autoRevealDependencies", bindings);
+      }
+      case "Axiom": {
+        return A("axiom");
+      }
+      case "Compile": {
+        return A1("compile", bindings);
+      }
+      case "Concurrent": {
+        return A("concurrent");
+      }
+      case "Contradiction": {
+        return A("contradiction");
+      }
+      case "DisableNonlinearArithmetic": {
+        return A1("disable-nonlinear-arithmetic", bindings);
+      }
+      case "Error": {
+        if (Get(bindings, 0, out var failureMessage) && failureMessage != null) {
+          if (Get(bindings, 1, out var successMessage) && successMessage != null) {
+            return A("error", failureMessage, successMessage);
+          }
+
+          return A("error", failureMessage);
+        }
+
+        return Error("@error requires at least one argument", atAttribute.Tok);
+      }
+      case "Extern": {
+        if (Get(bindings, 0, out var externName) && externName != null) {
+          if (Get(bindings, 1, out var moduleName) && moduleName != null) {
+            return A("extern", moduleName, externName);
+          }
+
+          return A("extern", externName);
+        }
+
+        return A("extern");
+      }
+      case "Focus": {
+        return A("focus");
+      }
+      case "Fuel": {
+        if (Get(bindings, 0, out var externName) && externName != null) {
+          if (Get(bindings, 1, out var moduleName) && moduleName != null) {
+            return A("extern", moduleName, externName);
+          }
+
+          return A("extern", externName);
+        }
+
+        return A("extern");
+      }
+      case "Id": {
+        return A1("id", bindings);
+      }
+      case "Induction": {
+        return A1("induction", bindings);
+      }
+      case "IsolateAssertions": {
+        return A("isolate_assertions");
+      }
+      case "NativeType": {
+        if (Get(bindings, 0, out var native) && native != null) {
+          if (native is LiteralExpr { Value: false }) {
+            return A("nativeType", native);
+          }
+
+          if (Get(bindings, 1, out var nameExpr) && nameExpr != null) {
+            return A("nativeType", nameExpr);
+          }
+        }
+
+        return A("nativeType");
+      }
+      case "Options": {
+        return A1("options", bindings);
+      }
+      case "Print": {
+        return A("print");
+      }
+      case "Priority": {
+        return A1("priority", bindings);
+      }
+      case "ResourceLimit": {
+        return A1("resource_limit", bindings);
+      }
+      case "SelectiveChecking": {
+        return A("selective_checking");
+      }
+      case "SplitHere": {
+        return A("split_here");
+      }
+      case "StartCheckingHere": {
+        return A("start_checking_here");
+      }
+      case "Subsumption": {
+        return A1("subsumption", bindings);
+      }
+      case "Synthesize": {
```

Comment:
Shouldn't the attributes be defined in a `.dfy` file?

Reply:
That can definitely be done in the future, and I already organized the code accordingly. For now, I'll keep the refactoring to a dafny file out of scope so all the code for the definition and the sync is in one place, and no pre-parsing and resolving is needed.
Moreover, two attributes (Induction and Trigger) don't have a typed counterpart so can't be expressed in Dafny.